ttnn/cpp/ttnn/tensor/layout/page_config.cpp

```diff
@@ -130,20 +130,21 @@ RowMajorPageConfig::RowMajorPageConfig(const Tile& tile) : tile_(tile) {}
 
 Alignment RowMajorPageConfig::create_default_alignment(DataType dtype, const MemoryConfig& memory_config) const {
     {
-        uint32_t width_alignment = 1;
         if (memory_config.shard_spec.has_value()) {
             const auto& shard_spec = memory_config.shard_spec.value();
-            if (shard_spec.physical_shard_shape.has_value()) {
-                return Alignment(shard_spec.physical_shard_shape.value());
+            if (shard_spec.mode == ShardMode::LOGICAL) {
+                if (shard_spec.physical_shard_shape.has_value()) {
+                    return Alignment(shard_spec.physical_shard_shape.value());
+                }
+                return Alignment({shard_spec.shape[1]});
```

Comment:
```
return shard_spec.physical_shard_shape.has_value()
      ? Alignment(*shard_spec.physical_shard_shape)
      : Alignment({shard_spec.shape[1]});
```

Reply:
Isn't this the same code?

Reply:
Yes, but it is more compact, layers of `if`s here are becoming hard to track.